sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -86,362 +95,373 @@ export class BatchingReceiver extends MessageReceiver {
-          actionAfterWaitTimeout,
-          (maxWaitTimeInSeconds as number) * 1000
-        );
-        // TODO: Disabling this for now. We would want to give the user a decent chance to receive
-        // the first message and only timeout faster if successive messages from there onwards are
-        // not received quickly. However, it may be possible that there are no pending messages
-        // currently on the queue. In that case waiting for idleTimeoutInSeconds would be
-        // unnecessary.
-        // There is a management plane API to get runtimeInfo of the Queue which provides
-        // information about active messages on the Queue and it's sub Queues. However, this adds
-        // a little complexity. If the first message was delayed due to network latency then there
-        // are bright chances that the management plane api would receive the same fate.
-        // It would be better to weigh all the options before making a decision.
-        // resetTimerOnNewMessageReceived();
-      };
-
-      if (!this.isOpen()) {
-        // clear detachedError since we are reconnecting.
-        this.detachedError = undefined;
-        log.batching(
-          "[%s] Receiver '%s', setting max concurrent calls to 0.",
-          this._context.namespace.connectionId,
-          this.name
-        );
-        // while creating the receiver link for batching receiver the max concurrent calls
-        // i.e. the credit_window on the link is set to zero. After the link is created
-        // successfully, we add credit which is the maxMessageCount specified by the user.
-        this.maxConcurrentCalls = 0;
-        const rcvrOptions = this._createReceiverOptions(false, {
-          onMessage: onReceiveMessage,
-          onError: onReceiveError,
-          onSessionError: onSessionError,
-          onSettled: onSettled,
-          onClose: onReceiveClose,
-          onSessionClose: onSessionClose
-        });
-        this._init(rcvrOptions)
-          .then(() => {
-            this._receiver!.on(ReceiverEvents.receiverDrained, onReceiveDrain);
-            addCreditAndSetTimer();
-            return;
-          })
-          .catch(reject);
-      } else {
-        addCreditAndSetTimer(true);
-        this._receiver!.on(ReceiverEvents.message, onReceiveMessage);
-        this._receiver!.on(ReceiverEvents.receiverError, onReceiveError);
-        this._receiver!.on(ReceiverEvents.receiverDrained, onReceiveDrain);
-        this._receiver!.session.on(SessionEvents.sessionError, onSessionError);
-      }
-    });
+      });
+    const config: RetryConfig<ServiceBusMessageImpl[]> = {
+      connectionHost: this._context.namespace.config.host,
+      connectionId: this._context.namespace.connectionId,
+      operation: receiveMessages,
+      operationType: RetryOperationType.receiveMessage,
+      abortSignal: undefined,
+      retryOptions: retryOptions
+    };
+    return retry<ServiceBusMessageImpl[]>(config);
```

Comment:
In the receiver.ts file, all the management operations have the call to retry() being done from the outside.
I wonder if we can move the above retry() call for `receiveMessages` to the receiver.ts file as well.
It won't make any functional difference, the only benefit is that it would be easier on the eyes when one reads through the `Receiver` class to find all the retry() calls being done there.

Thoughts?

Reply:
I did think about it, moved it now.